plugins/node/opentelemetry-koa-instrumentation/src/koa.ts

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { BasePlugin } from '@opentelemetry/core';
+import * as koa from 'koa';
+import * as shimmer from 'shimmer';
+import {
+  Parameters,
+  KoaMiddleware,
+  KoaContext,
+  KoaComponentName,
+  kLayerPatched,
+} from './types';
+import { VERSION } from './version';
+import { getMiddlewareMetadata } from './utils';
+
+/** Koa instrumentation for OpenTelemetry */
+export class KoaInstrumentation extends BasePlugin<typeof koa> {
+  static readonly component = KoaComponentName;
+  readonly supportedVersions = ['^2.0.0'];
+
+  constructor(readonly moduleName: string) {
+    super('@opentelemetry/koa-instrumentation', VERSION);
+  }
+
+  /**
+   * Patches Koa operations by wrapping the Koa.use function
+   */
+  protected patch(): typeof koa {
+    this._logger.debug('Patching Koa');
+
+    if (this._moduleExports === undefined || this._moduleExports === null) {
+      return this._moduleExports;
+    }
+    this._logger.debug('Patching Koa.use');
+    const appProto = this._moduleExports.prototype;
+    shimmer.wrap(appProto, 'use', this._getKoaUsePatch.bind(this));
```

Comment:
use of `bind` is often confusing
```suggestion
    shimmer.wrap(appProto, 'use', (original) => this._getKoaUsePatch(original));
```

Reply:
Not sure the bind is necessary at all

Reply:
Without the bind, if someone modifies the `_getKoaUsePatch` function later to reference `this`, it will fail and may not be obvious why.

Reply:
Hmm okay true - would you prefer that I leave it in? It is actually unnecessary for the current implementation

Reply:
@dyladan The `this` is correctly typed there https://github.com/open-telemetry/opentelemetry-js-contrib/pull/144/files#diff-58a707d20a03601b47963c43aae79ed7R66 so i think it's fine not binding it.

Reply:
That is only a hint to typescript telling it what the `this` object's type _should_ be. It does not change behavior at all. If you do not bind context properly, the typings will simply be wrong.

Reply:
Ah I see there is no reference into the plugin. Sorry for my hasty judgement. It is fine without bind or arrow.

Reply:
Okay, I have removed it